packages/@aws-cdk/core/README.md

```diff
@@ -1302,4 +1302,118 @@ permissions boundary attached.
 
 For more details see the [Permissions Boundary](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_iam-readme.html#permissions-boundaries) section in the IAM guide.
 
+## Policy Validation
+
+If you or your organization use any policy validation tool, such as
+[CloudFormation
+Guard](https://docs.aws.amazon.com/cfn-guard/latest/ug/what-is-guard.html) or
+[OPA](https://www.openpolicyagent.org/), to define constraints on your
+CloudFormation template, you can integrate them with the CDK at synthesis time.
+By using the appropriate plugin, you can make the CDK application check the
+generated CloudFormation template against your policies immediately after
+synthesis. If there are any violations, the synthesis will fail and a report
+will be printed to the console.
+
+### For application developers
+
+To use one or more validation plugins in your application, use the
+`validationPlugins` property of `Stage`:
+
+```ts
+// globally for the entire app (an app is a stage)
+const app = new App({
+  validationPlugins: [
+    // These hypothetical classes implement IValidationPlugin:
+    new ThirdPartyPluginX(), 
+    new ThirdPartyPluginY(),
+  ],
+});
+
+// only apply to a particular stage
+const prodStage = new Stage(app, 'ProdStage', {
+  validationPlugins: [...],
+});
+```
+
+Immediately after synthesis, all plugins registered this way will be invoked to
+validate all the templates generated in the scope you defined. In particular, if
+you register the templates in the `App` object, all templates will be subject to
+validation.
+
+By default, the report will be printed in a human readable format. If you want a
+report in JSON format, use the `synth()` method:
```

Comment:
Will this be configurable via CLI or will end users need to watch for env var to dynamically flip this? Also will the CDK behave correctly for machine consumers such that `cdk synth > results.json` will behave predicatbly?

Reply:
Ah. Will there be a simple way to extract this information? I'm thinking about CI environments where I might want to generate a violations report or Pull Request comment. Otherwise the ability to output as JSON isn't super helpful 😅

Reply:
@otaviomacedo it might be simple to just use a context variable instead of the parameter on `synth` that way you could enable it either via the CLI or via the app.

```
cdk synth --context @aws-cdk/core:validationReportJson=true
```

```ts
const app = new App({
  context: {
    '@aws-cdk/core:validationReportJson': true,
  },
});
```

>The report is printed to stderr

Could we report it to `stdout` if it is JSON?

Reply:
Good idea. Done.

Reply:
Thank you!